src/main/scala/org/orleans/silo/Services/Grain/GrainRef.scala

```diff
@@ -14,6 +14,13 @@ object GrainRef extends LazyLogging {
 
 }
 
+trait GrainReference {
+  var grainRef: GrainRef = _
+  var masterRef: GrainRef = _
+  def setGrainRef(grainRef: GrainRef) = this.grainRef = grainRef
+  def setMasterGrain(master: GrainRef) = this.masterRef = master
+}
+

```

Comment:
It does extend it right (at least GrainRef extends GrainReference)? I agree that having both GrainRef and GrainReference is a bit of a hassle, but I use this so that people can create their own interface around a GrainRef and make the life of a client even easier. Now you can do `runtime.createGrain[Twitter, TwitterRef]` and this will return the `TwitterRef` instantiated with the correct `GrainRef`.